File: app/src/main/java/com/example/ideal/myapplication/adapters/ServiceProfileAdapter.java
package com.example.ideal.myapplication.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ideal.myapplication.R;
import com.example.ideal.myapplication.adapters.foundElements.FoundServiceProfileElement;
import com.example.ideal.myapplication.fragments.objects.Service;

import java.util.ArrayList;

public class ServiceProfileAdapter extends RecyclerView.Adapter<ServiceProfileAdapter.ServiceProfileViewHolder> {

    private static final String TAG = "DBInf";

    private int numberItems;
    private ArrayList<Service> serviceList;
    private Context context;

    public ServiceProfileAdapter(int numberItems, ArrayList<Service> serviceList) {
        this.numberItems = numberItems;
        this.serviceList = serviceList;
    }

    @NonNull
    @Override
    public ServiceProfileViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        Log.d(TAG, "onCreateViewHolder: ");
        context = viewGroup.getContext();
        int layoutIdForListItem = R.layout.found_service_profile_element;
        //Класс, который позволяет создавать представления из xml файла
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        // откуда, куда, необходимо ли помещать в родителя
        View view = layoutInflater.inflate(layoutIdForListItem, viewGroup, false);
        return new ServiceProfileViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ServiceProfileViewHolder serviceProfileViewHolder, int i) {
            serviceProfileViewHolder.bind(serviceList.get(i));
    }

    @Override
    public int getItemCount() {
        return numberItems;
    }

    class ServiceProfileViewHolder extends RecyclerView.ViewHolder {

        private View view;

        ServiceProfileViewHolder(@NonNull View itemView) {
            super(itemView);
            view = itemView;
        }

        void bind(Service service) {
            Log.d(TAG, "bind: "  );
            FoundServiceProfileElement foundServiceProfileElement = new FoundServiceProfileElement(service,view,context);
            foundServiceProfileElement.createElement();
        }
    }
}

File: app/src/main/java/com/bunbeauty/ideal/myapplication/cleanArchitecture/adapters/ServiceProfileAdapter.kt
package com.bunbeauty.ideal.myapplication.cleanArchitecture.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.android.ideal.myapplication.R
import com.bunbeauty.ideal.myapplication.cleanArchitecture.adapters.foundElements.FoundServiceProfileElement
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.models.entity.Service
import com.bunbeauty.ideal.myapplication.cleanArchitecture.data.db.models.entity.User

class ServiceProfileAdapter(private val serviceList: MutableList<Service>, private val user: User) :
    RecyclerView.Adapter<ServiceProfileAdapter.ServiceProfileViewHolder>() {
    private lateinit var context: Context

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ServiceProfileViewHolder {
        context = viewGroup.context
        val layoutInflater = LayoutInflater.from(viewGroup.context)
        val view = layoutInflater.inflate(
            R.layout.found_service_profile_element,
            viewGroup, false
        )
        return ServiceProfileViewHolder(view)
    }

    override fun onBindViewHolder(serviceProfileViewHolder: ServiceProfileViewHolder, i: Int) {
        serviceProfileViewHolder.bind(serviceList[i], user)
    }

    override fun getItemCount(): Int {
        return serviceList.size
    }

    inner class ServiceProfileViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {

        fun bind(service: Service, user: User) {
            val foundServiceProfileElement = FoundServiceProfileElement(service, user, context)
            foundServiceProfileElement.createElement(view)
        }
    }

    companion object {
        private val TAG = "DBInf"
    }
}
